File: Uni/Python/readFasta.py
#!/usr/bin/env python3

import sys

sys.path.append("/Users/gabriel/Kurse/algorithmen/lib/python")
import fasta
#from fasta import Fasta


# Argument: Dateiname
try:
    fastaFile = sys.argv[1]
except:
    print("too few arguments ...\a\n")
    sys.exit(1)


# Datei lesen
seq = fasta.Fasta(fastaFile)

# Informationen
print("Habe " + str(seq.getNumSeq()) +
      " Sequenzen aus der Datei " + seq.getFilename() + " gelesen.")

print(seq.getAllIds())
print(seq.getNumSeq())
print(seq.getSequence("nf|0001f"))
seq.setSequence("rababer", "saft")
print(seq.getAllIds())
print(seq.getNumSeq())
seq.setSequence("rababer", "adads")
print(seq.getAllIds())
print(seq.getNumSeq())

seq.writeFasta("blubber.fasta", 80)

File: Uni/Perl/readFasta.pl
#!/usr/bin/perl -W

use strict;
use warnings;

use BioinfAlg2::ReadFastA;
use BioinfAlg2::FastA;

my $file;

if ($ARGV[0]) {
    $file = $ARGV[0];
}
else {
    print STDERR "too few arguments ...\a\n";
    exit 1;
}

#################################################################
# Einlesen das Fasta-Files
#################################################################

my $seqfile = BioinfAlg2::ReadFastA -> new;

my $fasta = $seqfile -> read_sequence($file);

my $seq = $fasta -> sequence;
my $ids = $fasta -> id;

printf("Eingabedatei: %s\n", $seqfile -> filename);

foreach (@$ids) {
    print $_."\n";
}
